library/components/TTooltip.vue

```diff
@@ -56,27 +62,31 @@ export default {
 
       const positions = { ...this.positions };
       if (triggerElement && tooltipElement) {
-        const position = triggerElement.getBoundingClientRect();
         if (this.position === "top") {
-          positions.tooltip.top = position.top - (tooltipElement.clientHeight + 10) + "px";
-          positions.tooltip.left = position.left - tooltipElement.clientWidth / 2 + "px";
-          positions.arrow.top = position.top - 20 + "px";
-          positions.arrow.left = position.left + "px";
+          positions.tooltip.bottom =
+            triggerElement.clientHeight +
+            tooltipElement.clientHeight +
+            10 +
+            "px";
+          positions.tooltip.right = "25px";
+          positions.arrow.bottom = "-6px";
+          positions.arrow.left = "25px";
         } else if (this.position === "bottom") {
-          positions.tooltip.top = position.top + triggerElement.clientHeight + 15 + "px";
-          positions.tooltip.left = position.left - tooltipElement.clientWidth / 2 + "px";
-          positions.arrow.top = position.top + triggerElement.clientHeight + 10 + "px";
-          positions.arrow.left = position.left + "px";
+          positions.tooltip.top = "12px";
+          positions.tooltip.right = "25px";
+          positions.arrow.top = "-6px";
+          positions.arrow.left = "25px";
         } else if (this.position === "left") {
-          positions.tooltip.top = position.top - tooltipElement.clientHeight / 2 + "px";
-          positions.tooltip.left = position.left - tooltipElement.clientWidth - 15 + "px";
-          positions.arrow.top = position.top + "px";
-          positions.arrow.left = position.left - 25 + "px";
+          positions.tooltip.top = "-20px";
+          positions.tooltip.left =
+            "-" + (tooltipElement.clientWidth - 7) + "px";
+          positions.arrow.right = "-6px";
+          positions.arrow.top = "6px";
         } else {
-          positions.tooltip.top = position.top - tooltipElement.clientHeight / 2 + "px";
-          positions.tooltip.left = position.left + triggerElement.clientWidth + 15 + "px";
-          positions.arrow.top = position.top + "px";
-          positions.arrow.left = position.left + triggerElement.clientWidth + 10 + "px";
+          positions.tooltip.top = "-20px";
+          positions.tooltip.left = triggerElement.clientWidth + 7 + "px";
+          positions.arrow.left = "-6px";
+          positions.arrow.top = "5px";
```

Comment:
Change 3:
More solid placement values around trigger, thanks to relative position of parent, this will always be accurate placement.